app/lib/features/chat/widgets/room_avatar.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class RoomAvatar extends ConsumerWidget {
+  final String roomId;
+  final double avatarSize;
+
+  const RoomAvatar({
+    required this.roomId,
+    this.avatarSize = 36,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    //Fetch room conversations details from roomId
+    return SizedBox(
+      width: avatarSize,
+      height: avatarSize,
+      child: ref.watch(chatProvider(roomId)).when(
+            data: (convo) => chatAvatarUI(convo, ref),
+            error: (e, s) => Center(child: Text('Loading room failed: $e')),
+            loading: () => const Center(child: Text('loading...')),
+          ),
+    );
+  }
+
+  Widget chatAvatarUI(Convo convo, WidgetRef ref) {
+    //Data Providers
+    final convoProfile = ref.watch(chatProfileDataProvider(convo));
+
+    //Manage Avatar UI according to the avatar availability
+    return convoProfile.when(
+      data: (profile) {
+        //Show conversations avatar if available
+        //Group : Show default image if avatar is not available
+        if (profile.hasAvatar() || !profile.isDm) {
+          return ActerAvatar(
+            uniqueId: roomId,
+            mode: profile.isDm ? DisplayMode.User : DisplayMode.Space,
+            displayName: profile.displayName ?? roomId,
+            avatar: profile.getAvatarImage(),
+            size: 18,
+          );
+        }
+
+        //Type == DM && Avatar not available : Handle avatar according to the members counts
+        else {
+          return dmAvatar(ref);
+        }
+      },
+      skipLoadingOnReload: false,
+      error: (err, stackTrace) {
+        debugPrint('Failed to load avatar due to $err');
+        return ActerAvatar(
+          uniqueId: convo.getRoomIdStr(),
+          mode: convo.isDm() ? DisplayMode.User : DisplayMode.GroupChat,
+          displayName: convo.getRoomIdStr(),
+          size: avatarSize,
+        );
+      },
+      loading: () => const CircularProgressIndicator(),
+    );
+  }
+
+  Widget dmAvatar(WidgetRef ref) {
+    final client = ref.watch(clientProvider);
+    final convoMembers = ref.watch(chatMembersProvider(roomId));
+    return convoMembers.when(
+      data: (members) {
+        int count = members.length;
+
+        //Show member avatar
+        if (count == 1) {
+          return memberAvatar(members[0], ref);
+        }
+
+        //Show opponent member avatar
+        else if (count == 2) {
```

Comment:
let's keep if and else if together please.
```suggestion
        } else if (count == 2) {
          //Show opponent member avatar
```